msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java

```diff
@@ -77,13 +87,23 @@ private static class HomeAccountFilter implements CacheRecordFilter {
 
         @Override
         public List<ICacheRecord> filter(@NonNull final List<ICacheRecord> records) {
+            final String methodTag = HomeAccountFilter.class.getSimpleName() + ":filter";
             final List<ICacheRecord> result = new ArrayList<>();
 
             for (final ICacheRecord cacheRecord : records) {
                 final String acctHomeAccountId = cacheRecord.getAccount().getHomeAccountId();
                 final String acctLocalAccountId = cacheRecord.getAccount().getLocalAccountId();
-                if (acctLocalAccountId != null && acctHomeAccountId.contains(acctLocalAccountId)) {
-                    result.add(cacheRecord);
+                try {
+                    if (acctHomeAccountId.contains(acctLocalAccountId)) {
+                        result.add(cacheRecord);
+                    }
+                } catch (final NullPointerException e) {
```

Comment:
I thought it could be cleaner if we want to add logging as well.
If we were to check if home account id and local account id are null in the conditional, then we would have to check them individually again to provide the logging statements (since it could be possible that both are null). I also could have moved the lines in the catch block to the beginning of the loop and implemented some sort of boolean flag (to avoid checking for null again) to be checked in the actual conditional, but I opted for a try/catch block because I thought it was better condensed.

Reply:
Here's the flag option I was thinking about. Is this preferred?:
`                boolean shouldProceed = true;
                if (acctHomeAccountId == null) {
                    Logger.warn(methodTag, "Home account id is null.");
                    shouldProceed = false;
                }
                if (acctLocalAccountId == null) {
                    Logger.warn(methodTag, "Local account id is null.");
                    shouldProceed = false;
                }
                if (shouldProceed && !acctHomeAccountId.contains(acctLocalAccountId)) {
                    result.add(cacheRecord);
                }`

Reply:
yes, how about boolean isNullOrEmpty = false?

Reply:
Addressed: f17222b57bd615f9654573a530aba9b5cd6aaadb